orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -39,12 +38,16 @@ def execution_session_run_forward(execution_session, onnx_model, gradient_accumu
         forward_inputs = C.OrtValueVector()
         forward_inputs.reserve(len(inputs))
         for input in inputs:
-            forward_inputs.push_back(to_dlpack(input), input.dtype == torch.bool)
+            forward_inputs.push_back(_utils._torch_tensor_to_dlpack(input), input.dtype == torch.bool)
 
         forward_outputs = C.OrtValueVector()
         # Run and return module outputs.
         execution_session.run_forward(forward_inputs, forward_outputs, state, gradient_accumulation_manager.cache)
-        user_outputs = gradient_accumulation_manager.extract_outputs_and_maybe_update_cache(forward_outputs)
+        device = _utils.get_device_from_inputs(inputs, None)
+        if len(inputs) > 0:
+            _utils._check_same_device(device, "Input argument to forward", *inputs)
```

Comment:
We already run this before `execution_session_run_forward` is invoked. Do we need to do it again here?

Reply:
no.